mPwQlHID: Implementation of LANScanner (v. 0.0.1)

DaianAiziatov opened the pull request:
In addition I've also create temporary view for checking LANScanner implementation.

DaianAiziatov committed:
Exclude carthge from .swiftlint.yml

DaianAiziatov committed:
MMLanScanner has been included

DaianAiziatov committed:
LANScanner v. 0.1.1

DaianAiziatov committed:
mPwQlHID: Implementation of LANScanner (v. 0.0.1)

pryaniq commented on TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h:
```diff
@@ -0,0 +1,20 @@
+//
+//  Created by Daian Aiziatov on 12/01/2019.
+//  Copyright © 2019 Daian Aiziatov. All rights reserved.
+//
```
Could you remove copyrights in the whole PR?

DaianAiziatov replied:
Done.

pryaniq commented on .swiftlint.yml:
```diff
@@ -67,5 +67,6 @@ nesting:
 
 excluded:
     - TVRemoteControl/Resources/R.generated.swift
+    - Carthage
```
I see a lot of SwiftLint warnings. Update SwiftLint and fix them, please :)

DaianAiziatov replied:
Fixed.

pryaniq commented on TVRemoteControl.xcodeproj/project.pbxproj:
```diff
@@ -379,8 +558,9 @@
-				DEVELOPMENT_TEAM = "";
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = WXMC8MD6TM;
```
Could you revert these changes? Since we don't have a development profile with permitted devices we mustn't commit these changes into common codebase after using a personal device.

DaianAiziatov replied:
Reverted.

pryaniq commented on TVRemoteControl.xcodeproj/project.pbxproj:
```diff
@@ -41,10 +89,118 @@
 			isa = PBXGroup;
 			children = (
 				1922112B21AFC39F00C0A68C /* Rswift.framework */,
+				2B1AC8592220FC88009BBBF1 /* MMLanScan */,
```
How I can see that's not Framework. I think better to move into a separated directory in the project. For instance, `TVRemoteControl/Third-Party/MMLanScan`

DaianAiziatov replied:
Moved to app folder.

pryaniq commented on TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h:
```diff
@@ -0,0 +1,20 @@
+#import "MMLANScanner.h"
+#import "MACOperation.h"
+#import "MacFinder.h"
+#import "MMDevice.h"
```
The biggest part of these imports is not used in the project. I tried to exclude them and everything works fine. Please remove redundant imports

```
#import "MMLANScanner.h"
//#import "LANProperties.h"
//#import "PingOperation.h"
#import "MMLANScanner.h"
//#import "MACOperation.h"
//#import "MacFinder.h"
#import "MMDevice.h"
```

DaianAiziatov replied:
Removed.

pryaniq commented on TVRemoteControl/ScannerTableViewController.swift:
```diff
@@ -0,0 +1,100 @@
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
```
Do you remember what we discussed about `MVC` approach with working with `View`? Let's arrange a workshop where we perform refactoring of this screen.

DaianAiziatov replied:
I am down to workshop. Agreed, that this controller is poorly designed, but it is just for testing 🐷 Although It might be reused 🤔

pryaniq commented on TVRemoteControl/ScannerTableViewController.swift:
```diff
@@ -0,0 +1,100 @@
+        return progressView
+    }()
+
+    private var connectedTVs = [SystemInfo]()
```
According to popular Swift style guides, `var connectedTVs: [SystemInfo] = []` looks more usual

[example](https://github.com/raywenderlich/swift-style-guide)

DaianAiziatov replied:
Linted

DaianAiziatov committed:
Remove copyright + Swiftformat

DaianAiziatov committed:
Remove CODE_SIGN_IDENTITY

DaianAiziatov committed:
Ignore DS_Store

DaianAiziatov committed:
Move MMLANScanner to app folder; Remove redundant imports

DaianAiziatov committed:
Lint default value for array

DaianAiziatov committed:
Make LANScanner more testable

DaianAiziatov committed:
Add unit test target

DaianAiziatov committed:
Add unit tests for LANScanner

DaianAiziatov committed:
Swiftformat

pryaniq commented on TVRemoteControl/Services/TVScanner/DataResponseError.swift:
```diff
@@ -0,0 +1,26 @@
+//
+//  DataResponseError.swift
+//  RemoteControlPhilips
+//
+//  Created by Daian Aiziatov on 13/01/2019.
+//  Copyright © 2019 Daian Aiziatov. All rights reserved.
+//
```
What about this file and 3 others? 👯‍♂️

pryaniq commented on TVRemoteControl.xcodeproj/project.pbxproj:
```diff
@@ -357,8 +673,9 @@
-				DEVELOPMENT_TEAM = "";
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = WXMC8MD6TM;
```
🙈

pryaniq commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Implement a service for scanning LAN IP addresses with validation for finding TV devices](https://trello.com/c/mPwQlHID/12-implement-a-service-for-scanning-lan-ip-addresses-with-validation-for-finding-tv-devices)

pryaniq committed:
mPwQlHID: Implementation of LANScanner (v. 0.0.1)

  - remove copyrights
  - fix project file
  - comment out unit tests
  - return MMLanScan lib
